packages/shared/src/utils/helpers/__tests__/chart-barrier-store.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { configure } from 'mobx';
+import { ChartBarrierStore } from '../chart-barrier-store';
+import { BARRIER_COLORS, BARRIER_LINE_STYLES, CONTRACT_SHADES, DEFAULT_SHADES } from '../../constants';
+import { CONTRACT_TYPES } from '../../contract';
+
+type TOnChartBarrierChange = (high: string, low?: string) => void;
+
+configure({ safeDescriptors: false });
+
+let mockedChartBarrierStore: ChartBarrierStore;
+const initialHigh = '1.234';
+const initialLow = '1.200';
+const newBarriersData = { high: '+0.12', low: '-0.12', title: 'new title' };
+const newBarriersValuesArray = Object.values(newBarriersData) as [string, string, string];
+
+beforeEach(async () => {
+    mockedChartBarrierStore = new ChartBarrierStore(initialHigh, initialLow, jest.fn(), {
+        not_draggable: true,
+        color: BARRIER_COLORS.BLUE,
+        line_style: BARRIER_LINE_STYLES.SOLID,
+        hideBarrierLine: false,
+        shade: DEFAULT_SHADES['2'],
+    });
+});
+
+describe('ChartBarrierStore', () => {
+    it('should be initialized with properties', () => {
+        expect(mockedChartBarrierStore.barrier_count).toEqual(2);
+        expect(mockedChartBarrierStore.default_shade).toEqual(
+            DEFAULT_SHADES[mockedChartBarrierStore.barrier_count as keyof typeof DEFAULT_SHADES]
+        );
+        expect(mockedChartBarrierStore.draggable).toEqual(false);
+    });
+    it('should have draggable as true when initialized with not_draggable: false', () => {
+        mockedChartBarrierStore = new ChartBarrierStore(initialHigh, undefined, jest.fn(), {
+            not_draggable: false,
+        });
+        expect(mockedChartBarrierStore.draggable).toEqual(true);
+    });
+    it('should have high_barrier as 0 when initialized with empty args', () => {
+        mockedChartBarrierStore = new ChartBarrierStore();
+        expect(mockedChartBarrierStore.high).toEqual(0);
+    });
+    describe('updateBarriers', () => {
```

Comment:
I didn't know that we can separate tests cases with describe, cool)
Thank you for sharing )